src/main/java/org/hobbit/core/components/AbstractTaskGenerator.java

```diff
@@ -89,7 +89,7 @@
     protected DataSender sender2EvalStore;
     protected DataReceiver dataGenReceiver;
 
-    protected QueueingConsumer consumer;
+    protected CustomConsumer consumer;
```

Comment:
Is this `consumer` used anywhere? Could you please check this, e.g., in the tests? It does not seem to be configured, so if it is not used, please mark it as `@deprecated`.